department-of-veterans-affairs/vets-website
# ARF 86022 Integrate Form 21a submit FE with BE

## Summary
- Update submitURL to `${environment.API_URL}/accredited_representative_portal/v0/form21a`,
- Remove submit key so that we are using default submit function
<img width="754" alt="Screenshot 2024-07-02 at 10 25 10 AM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/22990386/ed25cdb5-453a-4134-8751-701519115417">
<img width="772" alt="Screenshot 2024-07-02 at 10 25 22 AM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/22990386/9b79e056-32fa-4d48-9b92-b4ed450df9e9">
<img width="214" alt="Screenshot 2024-07-02 at 10 27 53 AM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/22990386/7fbfe5af-c6aa-46a5-923c-e0918eeaa83d">

- Update place of birth so that the data is successfully being submitted with the POST (Follow on issue to return to align with design: [Update Address fields to not require street and postal code#87156](https://app.zenhub.com/workspaces/accredited-representative-facing-team-65453a97a9cc36069a2ad1d6/issues/gh/department-of-veterans-affairs/va.gov-team/87156))
<img width="459" alt="Screenshot 2024-07-02 at 10 29 42 AM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/22990386/00a24b8d-2cf9-471d-b61a-a923e2325206">

## Related issue(s)

- [#86022 Integrate the Form21a submit frontend with the vets-api controller](https://app.zenhub.com/workspaces/accredited-representative-facing-team-65453a97a9cc36069a2ad1d6/issues/gh/department-of-veterans-affairs/va.gov-team/86022)
- [[ARF] Digital Form 21a #83627](https://app.zenhub.com/workspaces/accredited-representative-facing-team-65453a97a9cc36069a2ad1d6/issues/gh/department-of-veterans-affairs/va.gov-team/83627)

## Testing done

- What was occurring?
  - submitURL was not present
  - submit function was what came with the generator
  - submit was not working
  - place of birth wouldn't submit data even if submit was working
- Steps to verify:
  - Run e2e testing
  - Look at the POST
  - Verify the POST includes all the data that we would expect

## Screenshots

### Before
<img width="531" alt="Screenshot 2024-07-02 at 10 33 12 AM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/22990386/e9b78e11-c785-4d1c-ac06-49d28469827e">

### After
- Update place of birth so that the data is successfully being submitted with the POST
<img width="459" alt="Screenshot 2024-07-02 at 10 29 42 AM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/22990386/00a24b8d-2cf9-471d-b61a-a923e2325206">
  - Follow on issue to return to align with design: [Update Address fields to not require street and postal code#87156](https://app.zenhub.com/workspaces/accredited-representative-facing-team-65453a97a9cc36069a2ad1d6/issues/gh/department-of-veterans-affairs/va.gov-team/87156)

## Acceptance criteria

### Quality Assurance & Testing
